qiskit/synthesis/su4/siswap_decompose.py

```diff
@@ -0,0 +1,286 @@
+
+        Returns:
+            QuantumCircuit: Synthesized circuit.
+        """
+        if not approximate:
+            basis_fidelity = 1.0
+
+        u_decomp = TwoQubitWeylDecomposition(Operator(unitary))
+
+        x = u_decomp.a
+        y = u_decomp.b
+        z = u_decomp.c
+
+        A1 = Operator(u_decomp.K1r)
+        A2 = Operator(u_decomp.K1l)
+        B1 = Operator(u_decomp.K2r)
+        B2 = Operator(u_decomp.K2l)
+
+        p = np.array([x, y, z])
+
+        if abs(z) <= x - y + _EPS:
+            polytope_projection = p
+        else:
+            polytope_projection = find_min_point([v - p for v in _POLYTOPE]) + p
+
+        candidate_points = [
+            _ID,  # 0 applications
+            _SISWAP,  # 1 application
+            polytope_projection,  # 2 applications
+            p,  # 3 applications
+        ]
+
+        if _num_basis_uses is None:
+            expected_fidelities = [
+                (1 - average_infidelity(p, q)) * basis_fidelity**i
+                for i, q in enumerate(candidate_points)
+            ]
+            best_nbasis = int(np.argmax(expected_fidelities))  # tiebreaks with smallest
+        else:
+            best_nbasis = _num_basis_uses
+
+        p = candidate_points[best_nbasis]
+
+        circuit = _can_circuit(*p)
+        circuit = circuit.compose(B1.to_instruction(), [0], front=True)
+        circuit = circuit.compose(B2.to_instruction(), [1], front=True)
+        circuit = circuit.compose(A1.to_instruction(), [0])
+        circuit = circuit.compose(A2.to_instruction(), [1])
+
+        # FIXME: cleaner to track global phase during construction
+        i = np.where(~np.isclose(np.ravel(Operator(circuit).data), 0.0))[0][0]
+        phase_diff = cmath.phase(Operator(unitary).data.flat[i] / Operator(circuit).data.flat[i])
+        circuit.global_phase += phase_diff
+
+        return self._decomposer1q(circuit)
+
+
+def _can_circuit(x, y, z):
+    """
+    Find a circuit that implements the canonical gate
+    CAN(x, y, z) := RXX(x) . RYY(y) . RZZ(z)
+    """
+    circuit = QuantumCircuit(2)
+    if np.allclose([x, y, z], _ID):
+        pass
+    elif np.allclose([x, y, z], _SISWAP):
+        circuit.append(SiSwapGate(), [0, 1])
+    elif abs(z) <= x - y + _EPS:  # red region
+        V = _interleaving_single_qubit_gates(x, y, z)
+        v_decomp = TwoQubitWeylDecomposition(Operator(V))
+        can = _remove_pre_post(V, v_decomp)
+        circuit = circuit.compose(can, [0, 1])
+
+        # Due to the symmetry of Weyl chamber CAN(pi/4, y, z) ~ CAN(pi/4, y, -z)
+        # we might get a V operator that implements CAN(pi/4, y, -z) instead
+        # we catch this case and fix it by local gates.
+        if not np.isclose(v_decomp.c, z):
+            circuit = circuit.compose(XGate(), [0], front=True)
+            circuit = circuit.compose(ZGate(), [1], front=True)
+            circuit = circuit.compose(YGate(), [1])
+    else:
+        if z < 0:  # blue region
```

Comment:
The body of this `else` is just an `if/else` itself - you could make line 163 an `elif` and remove a level of indentation.